### PR TITLE
[REVIEW] Create nyc_taxi_years_correlation.ipynb

### PR DESCRIPTION
**Pull request by taureandyernv**
Adds new cuspatial PIP notebook that only depends on cuspatial. also shows a capability cuspatial allows that is unique to cuspatial due to speed and shows how to use cuspatial in a workflow.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by taureandyernv** [comment]
Please do a code review.  We will update the changelog and envs as well

**Comment by thomcom** [comment]
A simple and effective demonstration of the current `point_in_polygon` API. Thanks @taureandyernv !

**Comment by taureandyernv** [comment]
will fix the envs now

**Comment by thomcom** [comment]
@harrism @trxcllnt @cwharris Anyone else care to give this a glance as well for approval?

**Comment by trxcllnt** [comment]
@thomcom yeah, I'll pull it down and give it a look-see

**Comment by taureandyernv** [comment]
updated conda envs.  its your baby now :)

File: notebooks/nyc_taxi_years_correlation.ipynb [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# tzones = gpd.GeoDataFrame.from_file('tzones_lonlat.json')\n",
+    "# tzones.to_file('cu_taxi_zones.shp')"
```
Comment (trxcllnt) [inline notebooks/nyc_taxi_years_correlation.ipynb:127]: These shouldn't be commented out, right?
```suggestion
    "tzones = gpd.GeoDataFrame.from_file('tzones_lonlat.json')\n",
    "tzones.to_file('cu_taxi_zones.shp')"
```

File: notebooks/nyc_taxi_years_correlation.ipynb [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+    "taxi2016['DOLocationID'] = 264\n",
+    "for i in range(len(pip_iterations)-1):\n",
+    "    start = pip_iterations[i]\n",
+    "    end = pip_iterations[i+1]\n",
+    "    pickups = cuspatial.point_in_polygon(taxi2016['pickup_longitude'] , taxi2016['pickup_latitude'], taxi_zones[0][start:end], taxi_zones[1], taxi_zones[2]['x'], taxi_zones[2]['y'])\n",
+    "    dropoffs = cuspatial.point_in_polygon(taxi2016['dropoff_longitude'] , taxi2016['dropoff_latitude'], taxi_zones[0][start:end], taxi_zones[1], taxi_zones[2]['x'], taxi_zones[2]['y'])\n",
+    "    for j in pickups.columns:\n",
+    "        taxi2016['PULocationID'].loc[pickups[j]] = j\n",
+    "    for j in dropoffs.columns:\n",
+    "        taxi2016['DOLocationID'].loc[dropoffs[j]] = j"
```
Comment (trxcllnt) [inline notebooks/nyc_taxi_years_correlation.ipynb:201]: This takes 2min 28s for me. Is this expected?

Comment (taureandyernv) [reply notebooks/nyc_taxi_years_correlation.ipynb:201]: Yeah, it is expected.  It has to do with the GPU that you are using.  GV100s are the fastest.  My RTX8000 takes 3 minutes.

**Comment by thomcom** [comment]
rerun tests

**Pull request merged**